### PR TITLE
[script.audio.motherearth] 1.0

### PR DESCRIPTION
**Pull request by FlorianReiterer**
### Description

Music Addon to play Mother Earth Radio, a free High-Resolution FLAC / AAC 96k/24bits webradio
### Checklist:

- [x ] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x ] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [ x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: script.audio.motherearth/service.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    def onPlayBackStopped(self):
+        self.reset()
+
+
+
+if __name__ == '__main__':
+    player = Player()
+    monitor = xbmc.Monitor()
+    while not monitor.abortRequested():
+        if monitor.waitForAbort(0.1):
```
Comment (enen92) [inline script.audio.motherearth/service.py:165]: any reason for such a small sleep time? 1 second would be enough no?

Comment (enen92) [reply script.audio.motherearth/service.py:165]: please also check addon-checker errors and warnings:

```
WARN: http://github.com/FlorianReiterer/script.audio.motherearth redirects to https://github.com/FlorianReiterer/script.audio.motherearth
INFO: Image icon exists
ERROR: icon should be solid. It has transparency.
ERROR: icon should have either 256x256 or 512x512 but it has 300x300
INFO: Image fanart exists
ERROR: fanart should have either 1280x720 or 1920x1080 or 3840x2160 but it has 1024x1025
INFO: fanart file size is fine 289KB
WARN: Complex entry point. Check: script.py | Counted lines: 48 | Lines allowed: 15
WARN: Complex entry point. Check: service.py | Counted lines: 139 | Lines allowed: 15
```

Comment (FlorianReiterer) [reply script.audio.motherearth/service.py]: fixed the artwork, and changed the sleep time to 1:
`        if monitor.waitForAbort(1):`

struggle with the http / https github redirect, sorry, will sort that out.
also i try shortening the script.py: i move the contents of it to another python script file and just point script.py to there like:
```
import new.py
new.run()
```
 right?
..but i struggle with that also. would you mind keeping those extra lines for now, i will get me some help and fix this as soon as i can.

Comment (enen92) [reply script.audio.motherearth/service.py]: the entrypoint size warning is just a recomendation, not something mandatory. What happens is that kodi does not compile the entrypoint to bytecode, so it compiles it every time the plugin runs. If you store most of it on a module and just execute (like above), the ammount of code to compile is far smaller. But you can do this later or even ignore.

Regarding the redirect to https, please use the https link in your adon.xml (`<source>` element).

Comment (FlorianReiterer) [reply script.audio.motherearth/service.py]: thanks! https is fixed also now, i'll go with the longer code for a start.

**Comment by enen92** [comment]
`ERROR: icon should be solid. It has transparency.`

**Review by enen92 — APPROVED** [review]

**Pull request merged**